Src/Plugins/PlaintextSecrets/PlaintextSecrets.json

```diff
@@ -4,15 +4,40 @@
       "Name": "DoNotExposePlaintextSecrets",
       "Id": "SEC1001",
       "Level": "Error",
+      "DefaultNameRegex": "(?i)\\.(azure|bat|cmd|config|cpp|cs|cscfg|definitions|dtsx|ini|java|jsx?|json|keys|loadtest|m|md|php|properties|ps1|psm1|pubxml|py|resx|sample|sql|ste|swift|test|tsx?|txt|waz|xml)$",
       "Description": "Do not expose plaintext (or base64-encoded plaintext) secrets in versioned engineering content.",
       "Message": "'{0:scanTarget}' contains an apparent {1:encoding}'{2:secretKind}' secret.",
       "MatchExpressions": [
         {
           "SubId": "AzureDevOpsPersonalAccessToken",
-          "NameRegex": "(?i)\\.(azure|bat|cmd|config|cpp|cs|cscfg|definitions|dtsx|ini|java|jsx?|json|keys|loadtest|m|md|php|properties|ps1|psm1|pubxml|py|resx|sample|sql|ste|swift|test|tsx?|txt|waz|xml)$",
           "Base64DecodedContentLength": 52,
           "ContentsRegex": "\\b(?<fingerprint>[a-z2-7]{52})",
           "MessageArguments": { "secretKind": "Azure DevOps personal access token" }
+        },
+        {
+          "SubId": "FacebookOAuthToken",
+          "ContentsRegex": "(?i)facebook.{0,30}['\"\\s]([0-9a-f]{32})['\"\\s]",
+          "MessageArguments": { "secretKind": "Facebook OAuth token" }
+        },
+        {
+          "SubId": "TwitterOAuthToken",
+          "ContentsRegex": "(?i)twitter.{0,30}['\"\\s]([0-9a-zA-Z]{35,44})['\"\\s]",
+          "MessageArguments": { "secretKind": "Twitter OAuth token" }
+        },
+        {
+          "SubId": "GoogleOAuth",
+          "ContentsRegex": "\"client_secret\":\"([a-zA-Z0-9-_]{24})\"",
```

Comment:
>zA [](start = 52, length = 2)

use (?i) in this pattern and drop this to a-z0-9. Actually, let's please use a common pattern for this everywhere, and let's make it a-z0-9 or  0-9a-z, whichever you prefer. This consistency will help with readability. ALSO: we need to be absolutely sure that all these tokens actually permit a mixture of upper and lowercase alpha. I'm not %100 sure, so let's chase that down.

Reply:
validated locally and it matches with a client_secret that i have. The format is:
15 letters or numbers (upper or lower case) + "-" + 2 letters or numbers (upper or lower case) + "-" + 5 letters or numbers (upper or lower case).

---
In reply to: [541058036](https://github.com/microsoft/sarif-pattern-matcher/pull/20#discussion_r541058036) [](ancestors = 541058036)